public/js/pimcore/object/helpers/selectField.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+pimcore.registerNS('pimcore.object.helpers.selectField');
+
+/**
+ * @private
+ */
+pimcore.object.helpers.selectField = {
+    OPTIONS_PROVIDER_TYPE_CONFIGURE: 'configure',
+    OPTIONS_PROVIDER_TYPE_SELECT_OPTIONS: 'select_options',
+    OPTIONS_PROVIDER_TYPE_CLASS: 'class',
+
+    selectOptionsStore: null,
+
+    /**
+     * @param {Object} datax
+     * @param {Ext.grid.Panel} valueGrid
+     * @returns {[
+     *     Ext.form.field.ComboBox,
+     *     Ext.form.field.Text,
+     *     Ext.form.field.Text,
+     *     Ext.form.field.ComboBox
+     *     ]}
+     */
+    getOptionsProviderFields: function (datax, valueGrid) {
+        var selectOptionsSelector = Ext.create('Ext.form.field.ComboBox', {
+            fieldLabel: t('selectoptions'),
+            emptyText: '',
+            value: datax.optionsProviderData,
+            hidden: true,
+            valueField: 'id',
+            displayField: 'text',
+            editable: false,
+            forceSelection: true,
+            queryMode: 'local',
+            store: this.getSelectOptionsStore(),
+            listeners: {
+                change: function (comboBox, newValue) {
+                    optionsProviderClass.setValue('\\Pimcore\\Bundle\\CoreBundle\\OptionsProvider\\SelectOptionsOptionsProvider');
```

Comment:
can we handle this with `pimcore.settings.select_options_provider_class`? https://github.com/pimcore/admin-ui-classic-bundle/blob/42ae0d40aa43d163e1bde2aa46f623830a406063/src/Controller/Admin/IndexController.php#L293